lib/pacemaker/pcmk_sched_actions.c

```diff
@@ -1371,63 +1371,62 @@ reschedule_recurring(pe_resource_t *rsc, const char *task, guint interval_ms,
     pe__set_action_flags(op, pe_action_reschedule);
 }
 
+/*!
+ * \internal
+ * \brief Schedule a reload of a resource on a node
+ *
+ * \param[in] rsc   Resource to reload
+ * \param[in] node  Where resource should be reloaded
+ */
 static void
-ReloadRsc(pe_resource_t * rsc, pe_node_t *node, pe_working_set_t * data_set)
+schedule_reload(pe_resource_t *rsc, pe_node_t *node)
 {
-    GList *gIter = NULL;
     pe_action_t *reload = NULL;
 
-    if (rsc->children) {
-        for (gIter = rsc->children; gIter != NULL; gIter = gIter->next) {
-            pe_resource_t *child_rsc = (pe_resource_t *) gIter->data;
-
-            ReloadRsc(child_rsc, node, data_set);
+    // For collective resources, just call recursively for children
+    if (rsc->variant > pe_native) {
+        for (GList *iter = rsc->children; iter != NULL; iter = iter->next) {
+            schedule_reload((pe_resource_t *) iter->data, node);
         }
```

Comment:
This could be written as `g_list_foreach(rsc->children, (GFunc) schedule_reload, node)`

Reply:
updated commit

Reply:
Looks good to me.